module/parser/src/jsTest/kotlin/org/cru/godtools/shared/tool/parser/internal/UsesResources.js.kt

```diff
@@ -11,8 +12,8 @@ internal actual val UsesResources.TEST_XML_PULL_PARSER_FACTORY: XmlPullParserFac
     get() = object : JsXmlPullParserFactory() {
         override fun readFile(fileName: String) = try {
             val path = resourcesDir?.let { "$RESOURCES_ROOT/$it" } ?: RESOURCES_ROOT
-            Resource("$path/$fileName").readText()
+            Promise.resolve(Resource("$path/$fileName").readText())
         } catch (e: RuntimeException) {
-            null
+            Promise.resolve(null)
```

Comment:
Should we use `Promise.reject` here? Unless you don't want it to throw an error.

Reply:
Also, Promises in KT might be different than TS, so I might be way off

Reply:
In reality I might want to throw an exception here specific to the file being missing that I can then catch and process at an outer level. But for now that'll just remain as a future potential improvement.